ledger/src/blockstore.rs

```diff
@@ -3203,20 +3192,11 @@ impl Blockstore {
         }
 
         self.db.write(write_batch)?;
-
-        let mut last_root = self.last_root.write().unwrap();
-        if *last_root == std::u64::MAX {
-            *last_root = 0;
-        }
```

Comment:
@carllin - As the author of this > 4 year old piece of code, any insights?

Reply:
The removal looks okay to me too, but I'll admit I haven't yet figured out why it was added in the first place.

Reply:
Ha fair, I plan on giving Carl some time to respond / bugging him if not. If enough time passes, I'll probably push as is.

I looked through and couldn't find any previous use of `u64::MAX` with the roots column, so I'm semi-confident in my theory that it was considered to initialize `last_root`  with `.unwrap_or(std::u64::MAX)` instead of `0` in this line, but then it was decided to use `0` instead.
https://github.com/solana-labs/solana/blob/9f25f67e6016dc8bd2ff7a7455eacf4cecbbc318/ledger/src/blockstore.rs#L326
There is no commit history to back me up, just a hypothesis so hopefully Carl remembers these random 3 lines from 4 years ago 😅

Reply:
> RwLock was intended to be initialized as the sentinel value std::u64::MAX,

This sounds right 😄 

I think this is safe to remove. It seems outside of `pub fn last_root(&self)` that read lock isn't being taken to protect any critical sections